backend/routes/seatbid.js

```diff
@@ -21,38 +19,60 @@ const router = express.Router();
 router.post('/start', authenticateUser, async (req, res) => {
     const { passengerId, seatNumber, flightId } = req.body;
     const biddingId = `${flightId}-${seatNumber}`;
-    const expirationTime = Date.now() + 30 * 60 * 1000; // 30 minutes
 
-    const existingBidding = Array.from(activeBiddings.values()).find(
-        bidding => bidding.passengerId === passengerId && bidding.status === 'ACTIVE'
-    );
+    try {
+        const flight = await prisma.flight.findUnique({ where: { flightId } });
+        if (!flight) {
+            return res.status(404).json({ error: 'Flight not found' });
+        }
 
-    if (existingBidding) {
-        return res.status(400).json({ error: 'You already have an active bidding' });
-    }
+        const departureDateTime = new Date(flight.scheduledDepartureTime);
 
-    const newBidding = {
-        passengerId,
-        seatNumber,
-        flightId,
-        startTime: Date.now(),
-        expirationTime,
-        status: 'ACTIVE'
-    };
+        if (isNaN(departureDateTime.getTime())) {
```

Comment:
When and why would this happen?

Reply:
That's an extra check I put to prevent inconsistencies in the format of time.